apps/quickbooks_online/utils.py

```
@@ -444,18 +462,22 @@ def sync_departments(self):
+        departments_generator = self.connection.departments.get_all_generator(sync_after)
 
-        active_existing_departments = list(DestinationAttribute.objects.filter(attribute_type='DEPARTMENT', workspace_id=self.workspace_id, active=True).values_list('destination_id', flat=True))
+        if not is_sync_after_timestamp_enabled:
+            active_existing_departments = list(DestinationAttribute.objects.filter(attribute_type='DEPARTMENT', workspace_id=self.workspace_id, active=True).values_list('destination_id', flat=True))
 
         for departments in departments_generator:
             department_attributes = []
             for department in departments:
                 department_attributes.append({'attribute_type': 'DEPARTMENT', 'display_name': 'Department', 'value': department['FullyQualifiedName'], 'destination_id': department['Id'], 'active': department['Active']})
-                if department['Id'] in active_existing_departments:
+                if department['Id'] in active_existing_departments and not is_sync_after_timestamp_enabled:
```

Pylint (3.3.7)

[error] Possibly using variable 'active_existing_departments' before assignment

(E0606)